baskets.go

```diff
@@ -165,12 +165,13 @@ func (req *RequestData) Forward(client *http.Client, config BasketConfig, basket
 
 			if err != nil {
 				log.Printf("[error] failed to forward request: %s", err)
-			} else {
-				io.Copy(ioutil.Discard, response.Body)
-				response.Body.Close()
+				return &http.Response{}
 			}
+
+			return response
 		}
 	}
+	return &http.Response{}
```

Comment:
here probably too: HTTP 502, I'll accept the PR, and think about the concept for error handling during request proxing, try to alling it with [HTTP Status Codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#5xx_Server_errors)